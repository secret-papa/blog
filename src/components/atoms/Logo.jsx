import React from "react";

import Text from "./Text";

const Logo = () => {
  return (
    <Text bold component="h1" variant="title1">
      :D
    </Text>
  );
};

export default Logo;
