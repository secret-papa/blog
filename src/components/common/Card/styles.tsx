import styled from "styled-components";

export const Root = styled.div`
  width: 320px;
  background-color: #ffffff;
  box-shadow: rgba(0, 0, 0, 0.04) 0px 4px 16px 0px;
  border-radius: 4px;
`;

export const Header = styled.div``;

export const CoverImageWrap = styled.div`
  height: 168px;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const Body = styled.div`
  padding: 1rem;
`;

export const Title = styled.h4`
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: 0.25rem;
  word-break: break-word;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: var(--color-grey-700);
`;

export const Description = styled.p`
  display: -webkit-box;
  height: 3.9375rem;
  font-size: 0.875rem;
  line-height: 1.5;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--color-grey-600);
  margin-bottom: 1.5rem;
`;

export const Meta = styled.div`
  font-size: 0.75rem;
  color: var(--color-grey-500);
`;
