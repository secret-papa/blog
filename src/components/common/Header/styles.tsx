import styled from "styled-components";

export const Wrapper = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 64px;
`;

export const Header = styled.div`
  display: flex;
  width: 1024px;
`;
