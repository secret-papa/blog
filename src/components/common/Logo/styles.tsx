import styled from "styled-components";

// eslint-disable-next-line import/prefer-default-export
export const Text = styled.span`
  font-size: 1.4rem;
  color: #2e4f4f;
`;
