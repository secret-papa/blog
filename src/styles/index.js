export { default as resetStyle } from "./reset";
export { default as theme } from "./theme";
