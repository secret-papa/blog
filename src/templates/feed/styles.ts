import styled from "styled-components";

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  background-color: #f8f9fa;
`;

export const Body = styled.main`
  width: 1024px;
`;

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
`;
