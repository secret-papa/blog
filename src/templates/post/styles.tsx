import styled, { css } from "styled-components";
import { Link } from "gatsby";

export const Root = styled.div`
  margin-bottom: 3rem;
`;

export const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: var(--color-grey-900);
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 5.5rem;
`;

export const Article = styled.article`
  width: 768px;
`;

export const CoverImageWrapper = styled.div`
  margin-top: 2rem;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
`;

export const MarkdownWrapper = styled.div`
  margin: 5rem 0 6rem;
`;

export const MarkdownBody = styled.div`
  & ol,
  ul {
    list-style: decimal;
  }
`;

export const NavigationButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  gap: 1rem;
`;

const leftDirectionCss = css`
  text-align: left;
`;
const rightDirectionCss = css`
  text-align: right;
`;

type Align = "left" | "right";

export const NavigationButton = styled(Link)<{ align?: Align }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 0.5rem;
  width: 100%;
  height: 4rem;
  background-color: #f8f9fa;
  padding-left: 1rem;
  padding-right: 1rem;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 0px 4px 0px;
  border: none;
  cursor: pointer;
  ${props => (props.align === "right" ? rightDirectionCss : leftDirectionCss)}
`;

export const NavigationButtonCaption = styled.span`
  font-size: 0.75rem;
  color: var(--color-grey-600);
`;

export const NavigationButtonTitle = styled.h3`
  font-size: 1.125rem;
  font-weight: bold;
  color: var(--color-grey-700);
`;
