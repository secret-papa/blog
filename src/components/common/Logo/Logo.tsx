import React from "react";
import { Text } from "./styles";

const Logo = () => <Text>Secret Blog</Text>;

export default Logo;
