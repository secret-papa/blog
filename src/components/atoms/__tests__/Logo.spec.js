import React from "react";

import { renderWithTheme } from "@tests";
import { theme } from "@src/styles";
import Logo from "../Logo";

describe("<Logo />", () => {
  it("should show correctly", () => {
    const { container } = renderWithTheme(<Logo />, theme);
    expect(container).toMatchSnapshot();
  });
});
