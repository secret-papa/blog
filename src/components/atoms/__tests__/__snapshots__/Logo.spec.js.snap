// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should show correctly 1`] = `
.emotion-0 {
  color: #757575;
  font: bold 3.2rem/1 sans-serif;
  font-family: inherit;
}

<div>
  <h1
    class="emotion-0"
  >
    :D
  </h1>
</div>
`;
